cpp/src/arrow/acero/task_util.cc

```diff
@@ -278,8 +278,8 @@ Status TaskSchedulerImpl::ExecuteMore(size_t thread_id, int num_tasks_to_execute
       bool task_group_finished = false;
       Status status = ExecuteTask(thread_id, group_id, task_id, &task_group_finished);
       if (!status.ok()) {
-        // Mark the remaining picked tasks as finished
-        for (size_t j = i + 1; j < tasks.size(); ++j) {
+        // Mark the current and remaining picked tasks as finished
+        for (size_t j = i; j < tasks.size(); ++j) {
```

Comment:
This is needed for the serial execution path to correctly set the task count.

Reply:
So PostExecuteTask need execute the task even if failed?

Reply:
Yeah this is pretty much the whole point of this change.